lib/urlwatch/reporters.py

```diff
@@ -175,48 +166,98 @@ def _parts(self):
+                    yield '<span style="color:green">+{html}</span>'.format(html=mark_to_html(line[1:]))
+                elif line[0] == '-':
+                    yield '<span style="color:red">-{html}</span>'.format(html=mark_to_html(line[1:]))
+                elif line[0] == '@':
+                    yield '<span style="background-color:whitesmoke">+{line}</span>'.format(line=line)
+                else:
+                    yield line[0] + mark_to_html(line[1:])
+        else:
+            for line in unified_diff.splitlines():
+                if line[0] == '+':
+                    yield SafeHtml('<span style="color:green">{line}</span>').format(line=line)
+                elif line[0] == '-':
+                    yield SafeHtml('<span style="color:red">{line}</span>').format(line=line)
+                elif line[0] == '@':
+                    yield SafeHtml('<span style="background-color:whitesmoke">{line}</span>').format(line=line)
+                else:
+                    yield SafeHtml('{line}').format(line=line)
 
     def _format_content(self, job_state, difftype):
         if job_state.verb == 'error':
-            return SafeHtml('<pre style="text-color: red;">{error}</pre>').format(error=job_state.traceback.strip())
+            return SafeHtml('<pre style="white-space:pre-wrap;color:red;">{error}</pre>').format(error=job_state.traceback.strip())
 
         if job_state.verb == 'unchanged':
-            return SafeHtml('<pre>{old_data}</pre>').format(old_data=job_state.old_data)
+            return SafeHtml('<pre style="white-space:pre-wrap">{old_data}</pre>').format(old_data=job_state.old_data)
 
         if job_state.old_data in (None, job_state.new_data):
             return SafeHtml('...')
 
-        if difftype == 'table':
-            timestamp_old = email.utils.formatdate(job_state.timestamp, localtime=True)
-            timestamp_new = email.utils.formatdate(time.time(), localtime=True)
-            html_diff = difflib.HtmlDiff()
-            return SafeHtml(html_diff.make_table(job_state.old_data.splitlines(keepends=True),
-                                                 job_state.new_data.splitlines(keepends=True),
-                                                 timestamp_old, timestamp_new, True, 3))
-        elif difftype == 'unified':
+        if difftype == 'unified':
+            # Style should include 'padding-left:2.3em;text-indent:-2.3em' to have hanging indents of long lines
+            # but Gmail strips the 'text-indent:-2.3em' and end up with everything indented
             return ''.join((
-                '<pre>',
+                '<pre style="white-space:pre-wrap">',
                 '\n'.join(self._diff_to_html(job_state.get_diff())),
                 '</pre>',
             ))
+        elif difftype == 'table':
+            timestamp_old = email.utils.formatdate(job_state.timestamp, localtime=1)
+            timestamp_new = email.utils.formatdate(time.time(), localtime=1)
```

Comment:
`localtime=True`

Reply:
Missed that upgrade to legacy code; done.